### PR TITLE
update gcc to gcc 10 and support c++17

### PR DESCRIPTION
**Pull request by stdpain**
update brpc to 0.9.7
    update boost to 1.73
    remove third-party boost 1.54 for mysql

reference to #5393 

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If these changes need document changes, I have updated the document
- [] Any dependent changes have been merged

### CODE REVIEW

File: thirdparty/build-thirdparty.sh [hunk 1]
```diff
@@ -416,25 +417,24 @@ build_boost() {
     check_if_source_exist $BOOST_SOURCE
     cd $TP_SOURCE_DIR/$BOOST_SOURCE
 
-    echo "using gcc : doris : ${CXX} ; " > tools/build/src/user-config.jam
-    ./bootstrap.sh --prefix=$TP_INSTALL_DIR
-    ./b2 --toolset=gcc-doris link=static -d0 -j$PARALLEL --without-mpi --without-graph --without-graph_parallel --without-python cxxflags="-std=c++11 -fPIC -I$TP_INCLUDE_DIR -L$TP_LIB_DIR" install
+    ./bootstrap.sh --prefix=$TP_INSTALL_DIR     
+    ./b2 link=static runtime-link=static -j$PARALLEL --without-mpi --without-graph --without-graph_parallel --without-python cxxflags="-std=c++11 -g -fPIC -I$TP_INCLUDE_DIR -L$TP_LIB_DIR" install
```
Comment (yangzhg) [inline thirdparty/build-thirdparty.sh:421]: missing whitespace

File: thirdparty/build-thirdparty.sh [hunk 2]
```diff
@@ -471,7 +471,8 @@ build_brpc() {
     rm -rf CMakeCache.txt CMakeFiles/
     LDFLAGS="-L${TP_LIB_DIR} -static-libstdc++ -static-libgcc" \
     ${CMAKE_CMD} -G "${GENERATOR}" -v -DBUILD_SHARED_LIBS=0 -DCMAKE_INSTALL_PREFIX=$TP_INSTALL_DIR \
-    -DBRPC_WITH_GLOG=ON -DWITH_GLOG=ON -DCMAKE_INCLUDE_PATH="$TP_INSTALL_DIR/include" \
+    -DBRPC_WITH_GLOG=ON -DWITH_GLOG=ON -DGFLAGS_LIBRARY=$TP_INSTALL_DIR/lib/libgflags.a -DGLOG_LIB=$TP_INSTALL_DIR/lib \
+    -DGFLAGS_INCLUDE_DIR=$TP_INSTALL_DIR/include -DGLOG_LIB=$TP_INSTALL_DIR/lib/libglog.a -DCMAKE_INCLUDE_PATH="$TP_INSTALL_DIR/include" \
     -DPROTOBUF_PROTOC_EXECUTABLE=$TP_INSTALL_DIR/bin/protoc .. 
     ${BUILD_SYSTEM} -j$PARALLEL && ${BUILD_SYSTEM} install
```
Comment (yangzhg) [inline thirdparty/build-thirdparty.sh:477]: missing whitespace

File: be/CMakeLists.txt [hunk 3]
```diff
@@ -130,11 +130,12 @@ endif()
...
+set(Boost_NO_BOOST_CMAKE OFF)
 
 if (NOT APPLE)
-    find_package(Boost 1.55.0 REQUIRED COMPONENTS thread regex filesystem system date_time program_options)
+    find_package(Boost 1.73.0 REQUIRED COMPONENTS regex system filesystem thread date_time program_options)
 else()
-    find_package(Boost 1.55.0 COMPONENTS thread regex filesystem system date_time program_options)
+    find_package(Boost 1.73.0 COMPONENTS thread regex system filesystem date_time program_options)
 endif()
 include_directories(${Boost_INCLUDE_DIRS})
```
Comment (yangzhg) [inline be/CMakeLists.txt:140]: this line is unuseless

File: be/CMakeLists.txt [hunk 4]
```diff
@@ -130,11 +130,12 @@ endif()
...
 message(STATUS ${Boost_LIBRARIES})
```
Comment (yangzhg) [inline be/CMakeLists.txt:141]: this line shoud removed when use find_package boost

Comment (stdpain) [reply be/CMakeLists.txt:141]: I will update cmake to cmake-3.19

File: be/CMakeLists.txt (hunk 3 above)
Comment (stdpain) [reply be/CMakeLists.txt]: Fixed

File: be/CMakeLists.txt (hunk 4 above)
Comment (stdpain) [reply be/CMakeLists.txt]: Fixed

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**